kernel/src/scan/mod.rs

```diff
@@ -160,7 +160,7 @@ impl ScanResult {
 /// store the name of the column, as that's all that's needed during the actual query. For
 /// `Partition` we store an index into the logical schema for this query since later we need the
 /// data type as well to materialize the partition column.
-#[derive(PartialEq, Debug)]
+#[derive(Clone, PartialEq, Debug)]
```

Comment:
Ah good point. The reason is really that I don't want to hold a reference to `all_fields` in `TableChangesScan` because the iterator needs to be free from lifetimes.

But really all I need is to `Arc` it. Now we don't need to clone the Vec.

Note that the existing `Scan::execute` still borrows self
```
    pub fn execute(
        &self,
        engine: Arc<dyn Engine>,
    ) -> DeltaResult<impl Iterator<Item = DeltaResult<ScanResult>> + '_> {
```
as you can see with that anonymous lifetime `'_`. I was planning on opening a followup PR to take that out too.